backend/src/controllers/BuildingController.js

```diff
@@ -0,0 +1,33 @@
+export class BuildingController {
+
+    constructor({ResourceService,BuildingService,ResourceRepo,BuildingRepo,db,errorCodes}) {
+        this.building = new BuildingService({BuildingRepo,ResourceService,ResourceRepo,db,errorCodes});
```

Comment:
@gulyaasjaanos a building service instance should be the dependency not the class, the whole point is to get rid of the logic of how to create an instance of the service here 
```suggestion
    constructor(buildingService) {
        this.buildingService = buildingService;
```
also the error codes can be imported as before, it's not needed to be replaced, they're constants